### PR TITLE
Changes to climber needed.

### PR DESCRIPTION
**Pull request by Beckoe**
Make changes per my comments.

### CODE REVIEW

**Comment by Beckoe** [comment]
This needs to be restructured entirely. Make it so one button press will move the motors to a set position and another button press will move them to a different set position. Make sure to use a PID controller when doing this. Invert one of the climber motors and have it follow the other motor. Constants do not need to be side-specific, simplify them to be used for both motors.

**Comment by Beckoe** [comment]
Also, talk to Owen about implementing a ProfiledPIDController.

File: RobotCode/src/main/java/frc/robot/commands/ClimberOverrideCommand.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+public class ClimberOverrideCommand extends Command
+{
+	private ClimberSubsystem climber;
+
+	public ClimberOverrideCommand(ClimberSubsystem climber)
+	{
+		this.climber = climber;
+		addRequirements(climber);
+		climber.setOverride(true);
```
Comment (renatodellosso) [inline RobotCode/src/main/java/frc/robot/commands/ClimberOverrideCommand.java:14]: Only actually set the override in Instantiate. We often create commands before we want them to run.

File: RobotCode/src/main/java/frc/robot/commands/ClimberToPositionCommand.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import frc.robot.subsystems.ClimberSubsystem;
+
+public class ClimberToPositionCommand extends Command
+{
+
+	public ClimberToPositionCommand(ClimberSubsystem climber, double targetPosition)
+	{
+		addRequirements(climber);
+
+		climber.setPosition(targetPosition);
```
Comment (renatodellosso) [inline RobotCode/src/main/java/frc/robot/commands/ClimberToPositionCommand.java:13]: Do this in Instantiate

File: RobotCode/src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 3]
```diff
@@ -1,63 +1,105 @@
...
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.lib.core.motors.TeamTalonFX;
 import frc.robot.constants.ClimberConstants;
 import frc.robot.constants.Ports;
 
 public class ClimberSubsystem extends SubsystemBase
 {
+	private Pigeon2 gyro;
+	private TeamTalonFX extendMotorLeft, extendMotorRight;
+	private double targetPosition;
```
Comment (renatodellosso) [inline RobotCode/src/main/java/frc/robot/subsystems/ClimberSubsystem.java:18]: Compress all the double declarations here into one line

**Review by renatodellosso — CHANGES_REQUESTED** [review]
Great work! Just a few small changes.

File: RobotCode/src/main/java/frc/robot/commands/RotateShooterMountToPositionCommand.java [hunk 4]
```diff
@@ -8,48 +8,48 @@
...
-    }
-
-    @Override
-    public boolean isFinished()
-    {
-        // We do the weird static access to avoid going through the variable
-        return Math.abs(ShooterMountSubsystem.getCurrentRotation()
-                - getPosition.getAsDouble()) < frc.robot.subsystems.ShooterMountSubsystem.DEADBAND;
-    }
+	private final ShooterMountSubsystem ShooterMountSubsystem;
```
Comment (PPenguin1) [inline RobotCode/src/main/java/frc/robot/commands/RotateShooterMountToPositionCommand.java:11]: Is this an error or is there something wrong with the code?

**Review by PPenguin1 — APPROVED** [review]

File: RobotCode/src/main/java/frc/robot/commands/RotateShooterMountToPositionCommand.java (hunk 4 above)
Comment (seankid) [reply RobotCode/src/main/java/frc/robot/commands/RotateShooterMountToPositionCommand.java:11]: *insert gaslighting here*

**Review by renatodellosso — APPROVED** [review]

**Pull request merged**